shared/src/tx_pool/pool.rs

```diff
@@ -136,6 +136,21 @@ impl TxPool {
         self.orphan.add_tx(cycles, size, tx, unknowns.into_iter())
     }
 
+    #[cfg(feature = "test")]
```

Comment:
I think we don't need these dirty methods to insert transactions with fake short id. There's an easy way to test collisions.

- Generate a transaction T, and add that transaction to the proposed pool, as usual.
- Generate a block, which includes the transaction. Change the merkle root to other value.
- Send the block as compact block, which does not prefill T.

The test should work because from the peer's perspective, it cannot tell the differences between a collision and a real unmatched merkle root.

Reply:
thanks.